foqus_lib/framework/graph/node.py

```diff
@@ -485,6 +592,26 @@ def setSim(self, newType=None, newModel=None, force=False, ids=None):
                         desc=item["description"],
                         optSet=NodeOptionSets.SINTER_OPTIONS,
                     )
+        elif self.modelType == nodeModelTypes.MODEL_ML_AI:
+            # link to pymodel class for ml/ai models
+            cwd = os.getcwd()
+            os.chdir(os.path.join(os.getcwd(), "user_ml_ai_models"))
+            try:  # see if custom layer script exists
+                module = import_module(str(self.modelName))  # contains CustomLayer
+                self.model = load(
+                    str(self.modelName) + ".h5",
+                    custom_objects={
+                        str(self.modelName): getattr(module, str(self.modelName))
+                    },
+                )
+            except:  # try to load model without custom layer
+                self.model = load(str(self.modelName) + ".h5")
+            os.chdir(cwd)  # reset to original working directory
```

Comment:
Resetting the working dir inside a `finally` block ensures that it is always restored, regardless of what exceptions are thrown in the previous lines.

```suggestion
            finally:
                os.chdir(cwd)  # reset to original working directory
```